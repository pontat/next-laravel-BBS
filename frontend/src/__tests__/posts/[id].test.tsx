import { render } from '@testing-library/react'
import Page from '../../pages/posts/[id]'

test('詳細ページ', () => {
  const { container } = render(
    <Page
      post={{
        id: 1,
        title: 'テストタイトル',
        content: 'テスト内容',
        created_at: '2021.01.01',
        updated_at: '2021.01.01',
        comments: [
          { id: 1, post_id: 1, content: 'テストコメント1', created_at: '2021.01.01', updated_at: '2021.01.01' },
          { id: 2, post_id: 1, content: 'テストコメント2', created_at: '2021.01.01', updated_at: '2021.01.01' },
          { id: 3, post_id: 1, content: 'テストコメント3', created_at: '2021.01.01', updated_at: '2021.01.01' },
        ],
      }}
    />
  )
  expect(container).toMatchSnapshot()
})
