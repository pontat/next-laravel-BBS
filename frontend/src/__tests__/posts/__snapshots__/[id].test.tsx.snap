// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`詳細ページ 1`] = `
<div>
  <h3
    class="text-lg font-bold leading-6 text-gray-900"
  >
    詳細
  </h3>
  <div
    class="mt-5 shadow rounded overflow-hidden"
  >
    <div
      class="p-4 bg-white space-y-6"
    >
      <div>
        <div
          class="text-sm font-bold text-gray-900"
        >
          タイトル
        </div>
        <div
          class="text-sm text-gray-900"
        >
          テストタイトル
        </div>
      </div>
      <div>
        <div
          class="text-sm font-bold text-gray-900"
        >
          内容
        </div>
        <div
          class="text-sm text-gray-900 whitespace-pre-wrap"
        >
          テスト内容
        </div>
      </div>
      <div>
        <div
          class="text-sm font-bold text-gray-900"
        >
          投稿日
        </div>
        <div
          class="text-sm text-gray-900"
        >
          2021.01.01
        </div>
      </div>
    </div>
    <div
      class="flex px-4 py-3 bg-gray-100"
    >
      <a
        class="px-4 py-2 bg-gray-600 hover:bg-gray-500 text-sm font-bold text-white rounded shadow hover:shadow-md cursor-pointer"
        href="/"
      >
        戻る
      </a>
    </div>
  </div>
  <div
    class="mt-5 shadow rounded overflow-hidden"
  >
    <div
      class="p-4 bg-white space-y-6"
    >
      <div>
        <label
          class="text-sm font-bold text-gray-900"
          for="comment"
        >
          コメント
        </label>
        <input
          class="w-full p-3 bg-white text-sm border border-blueGray-300 rounded"
          id="comment"
          placeholder="コメントを入力"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="flex px-4 py-3 bg-gray-100"
    >
      <button
        class="px-4 py-2 bg-blue-600 hover:bg-blue-500 text-sm font-bold text-white rounded shadow hover:shadow-md"
      >
        コメントする
      </button>
    </div>
  </div>
  <div
    class="mt-5 bg-white shadow rounded overflow-hidden"
  >
    <div
      class="p-4 bg-white"
    >
      <p
        class="text-sm text-gray-900"
      >
        テストコメント1
      </p>
    </div>
    <div
      class="px-4 py-3 bg-gray-100"
    >
      <p
        class="text-sm font-medium text-gray-500"
      >
        2021.01.01
      </p>
    </div>
  </div>
  <div
    class="mt-5 bg-white shadow rounded overflow-hidden"
  >
    <div
      class="p-4 bg-white"
    >
      <p
        class="text-sm text-gray-900"
      >
        テストコメント2
      </p>
    </div>
    <div
      class="px-4 py-3 bg-gray-100"
    >
      <p
        class="text-sm font-medium text-gray-500"
      >
        2021.01.01
      </p>
    </div>
  </div>
  <div
    class="mt-5 bg-white shadow rounded overflow-hidden"
  >
    <div
      class="p-4 bg-white"
    >
      <p
        class="text-sm text-gray-900"
      >
        テストコメント3
      </p>
    </div>
    <div
      class="px-4 py-3 bg-gray-100"
    >
      <p
        class="text-sm font-medium text-gray-500"
      >
        2021.01.01
      </p>
    </div>
  </div>
</div>
`;
